packages/jest-jasmine2/src/index.js

```diff
@@ -293,18 +293,15 @@ function jasmine2(
     }
   });
 
-  // extend jasmine matchers with `jest-matchers`
-  const jasmineExpect = env.expect;
-  env.expect = actual => {
-    const jasmineMatchers = jasmineExpect(actual);
-    const jestMatchers = expect(actual);
-    const not = Object.assign(jasmineMatchers.not, jestMatchers.not);
-    return Object.assign(jasmineMatchers, jestMatchers, {not});
-  };
-
   const snapshotState = snapshot.getSnapshotState(jasmine, testPath);
 
   env.addReporter(reporter);
+
+  // `jest-matchers` should be required inside test environment (vm).
+  // Otherwise if they throw, the `Error` class will differ from the `Error`
+  // class of the test and `error instanceof Error` will return `false`.
+  runtime.requireModule(path.resolve(__dirname, './extend_jasmine_expect.js'));
```

Comment:
this file was included in the coverage report and broke the coverage test. But in real world environment it'll most likely be excluded because it will be under `node_modules`